src/dataaggregator/aggregator/aggregator_test.go

```diff
@@ -97,55 +97,95 @@ var _ = Describe("Aggregator", func() {
 		var triggerMap map[string]*Trigger
 		var appChan chan *AppMonitor
 		var appMonitor *AppMonitor
-		JustBeforeEach(func() {
+		BeforeEach(func() {
 			appChan = make(chan *AppMonitor, 1)
 			aggregator = NewAggregator(logger, clock, TestPolicyPollerInterval, database, metricServer.URL(), TestMetricPollerCount)
-			triggerMap = map[string]*Trigger{testAppId: &Trigger{
-				AppId: testAppId,
-				TriggerRecord: &TriggerRecord{
-					InstanceMaxCount: 5,
-					InstanceMinCount: 1,
-					ScalingRules: []*ScalingRule{&ScalingRule{
-						MetricType:         "MemoryUsage",
-						StatWindowSecs:     300,
-						BreachDurationSecs: 300,
-						CoolDownSecs:       300,
-						Threshold:          30,
-						Operator:           "<",
-						Adjustment:         "-1",
-					}}},
-			}}
+
+		})
+		Context("when there are data in triggerMap", func() {
+			JustBeforeEach(func() {
+				triggerMap = map[string]*Trigger{testAppId: &Trigger{
```

Comment:
Why isn't this now in JustBeforeEach? Seems it should be in the BeforeEach and any test can override the triggerMap in their BeforeEach.
The JustBeforeEach can then just be the ConsumeTrigger call.